ctx.go

```diff
@@ -137,7 +137,9 @@ func newSpan(ctx context.Context, f *Func, args []interface{},
 
 		trace.decrementSpans()
 
-		if observer != nil {
+		// Re-fetch the observer, in case the value has changed since newSpan
+		// was called
+		if observer := trace.getObserver(); observer != nil {
```

Comment:
I thought about it and I think there's a problem doing this. If someone registered observer A when the process first started and the `observer.Start` in called when a span is newly created. Then another observer is registered later. The code right now will call `Finish` from the new observer but that may cause problem since the expectation would be calling `Start` and ` Finish` from the same observer.
Maybe we can do something like this?
 `if observer != nil; observer := trace.getObserver(); observer != nil { observer.Finish() }`

Reply:
I'm not quite sure what the problem is that you're referring to. Observer A would still have finished called on it, because when It finishes It calls finish on all the observers.

It's true that perhaps observer A is removed before finishes called, and observer B is the one where finish is called, but I don't see that as a problem.

My point is, if you add a second observer later the first observer is still called.

Reply:
Riiiight. Duh, I totally forgot that each registered observer will be called. I think I'm just worried that the Start won't be called for some oberver but the Finish always will.

Reply:
That is correct, but that is the known behavior if you register after the
spans have already started. We can't call start because that isn't when
they are actually starting, but in some use cases such as ours this is
completely fine behavior.